src/metrics/distances.ts

```diff
@@ -1,190 +1,156 @@
+  totalDistance: number;
+  maxSingleGameTotal: number;
+  // This has arrays to account for ties. Postions connect link to uas
+  pieceMaxes: {
+    distance: number;
+    uasArray: UASymbol[];
+    linkArray: string[];
+  };
+  gamesProcessed: number;
 
-    let distance = 0;
+  constructor() {
+    this.clear();
+  }
 
-    // Check if the move is a castling move
-    if (moveInfo.move.flags === 'k' || moveInfo.move.flags === 'q') {
-      let movingKing = 'k';
-      let movingRook, rookDistance;
+  clear(): void {
+    this.distanceMap = createUAPMap({
+      total: 0,
+      maxSingleGame: 0,
+    });
+    this.pieceWithHighestAvg = undefined;
+    this.pieceWithLowestAvg = undefined;
+    this.maxAvgDistance = 0;
+    this.minAvgDistance = Infinity; // Set high so first will overwrite
+    this.totalDistance = 0;
+    this.pieceMaxes = {
+      distance: 0,
+      uasArray: [],
+      linkArray: [],
+    };
+    this.gamesProcessed = 0;
+  }
 
-      if (moveInfo.move.flags === 'k') {
-        rookDistance = 2;
-        movingRook = 'rh';
-      } else {
-        rookDistance = 3;
-        movingRook = 'ra';
+  aggregate() {
+    // reset just the total aggregates, the others don't need a reset
+    this.totalDistance = 0;
+    for (const uas of Object.keys(this.distanceMap)) {
+      // increment the total distance by adding all pieces
+      this.totalDistance += this.distanceMap[uas].total;
+
+      const avgDistance = this.distanceMap[uas].total / this.gamesProcessed;
+      if (avgDistance > this.maxAvgDistance) {
+        this.maxAvgDistance = avgDistance;
+        this.pieceWithHighestAvg = uas as UASymbol;
       }
-
-      if (moveInfo.move.color === 'w') {
-        movingRook = movingRook.toUpperCase();
-        movingKing = movingKing.toUpperCase();
+      if (avgDistance < this.minAvgDistance) {
+        this.minAvgDistance = avgDistance;
+        this.pieceWithLowestAvg = uas as UASymbol;
       }
-
-      distanceMap[movingKing] += 2;
-      distanceMap[movingRook] += rookDistance;
-      singleGameDistanceTotal += 2;
-      singleGameDistanceTotal += rookDistance;
-    } else {
-      // Calculate the file (column) distance by subtracting ASCII values
-      const fileDist = Math.abs(fromMove.charCodeAt(0) - toMove.charCodeAt(0));
-      // Calculate the rank (row) distance by subtracting numeric values
-      const rankDist = Math.abs(Number(fromMove[1]) - Number(toMove[1]));
-      // The distance moved is the maximum of fileDist and rankDist
-      distance = Math.max(fileDist, rankDist);
-
-      distanceMap[move.uas] += distance;
-      singleGameDistanceTotal += distance;
-    }
-
-    if (distanceMap[move.uas] > maxDistance) {
-      maxDistance = distanceMap[move.uas];
-      maxDistancePiece = move.uas;
     }
   }
 
-  return {
-    maxDistancePiece,
-    maxDistance,
-    distanceMap,
-    singleGameDistanceTotal,
-  };
-}
+  logResults(): void {
+    console.timeEnd('Task 2: getMoveDistanceSetOfGames');
```

Comment:
this needs to be changed as this function name is obsolete?

Reply:
deleted